### PR TITLE
Update Inlet version to remove shared_ptr

### PR DESCRIPTION
**Pull request by joshessman-llnl**
Inlet was recently updated to not rely on shared pointers, meaning that we can remove all the messy deferencing when accessing with `operator[]`, for example.

The API should *hopefully* be more stable from here on out, so we can start look at extending inlet to the other modules.

### CODE REVIEW

File: src/drivers/serac.cpp [hunk 1]
```diff
@@ -83,12 +83,12 @@ int main(int argc, char* argv[])
   datastore.getRoot()->save("serac_input.json", "json");
 
   // Build the mesh
-  auto mesh_info      = (*inlet)["main_mesh"].get<serac::mesh::InputInfo>();
+  auto mesh_info      = inlet["main_mesh"].get<serac::mesh::InputInfo>();
```
Comment (jamiebramwell) [inline src/drivers/serac.cpp:86]: 👍

**Review by jamiebramwell — APPROVED** [review]
Looks good!

**Comment by jamiebramwell** [comment]
LGTM

**Review by white238 — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
Ran into an issue building TPLs on LC so the LC hostconfigs haven't been updated yet

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**